File: middle_level_bat/sample_app_001.py
# sample_app_001.py
import sys

if len(sys.argv) < 2:
    print('ファイル名を引数で指定してください。')
    sys.exit(1)
filename = sys.argv[1]
try:
    with open(filename, encoding='utf-8') as f:
        lines = f.readlines()
    line_count = len(lines)
    word_count = sum(len(line.strip().split()) for line in lines if line.strip())
    print(f'行数: {line_count}')
    print(f'単語数: {word_count}')
except Exception as e:
    print(f'ファイルの読み込みに失敗しました: {e}')

File: middle_level_bat/SampleApp001.java
import java.io.*;

public class SampleApp001 {
    public static void main(String[] args) {
        if (args.length < 1) {
            System.out.println("ファイル名を引数で指定してください。");
            return;
        }
        String filename = args[0];
        int lineCount = 0;
        int wordCount = 0;
        try (BufferedReader br = new BufferedReader(new InputStreamReader(new FileInputStream(filename), "UTF-8"))) {
            String line;
            while ((line = br.readLine()) != null) {
                lineCount++;
                wordCount += line.trim().isEmpty() ? 0 : line.trim().split("\\s+").length;
            }
            System.out.println("行数: " + lineCount);
            System.out.println("単語数: " + wordCount);
        } catch (IOException e) {
            System.out.println("ファイルの読み込みに失敗しました: " + e.getMessage());
        }
    }
}
